src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php

```diff
@@ -92,8 +104,22 @@ public function handleHydrate(AbstractMappingEvent $event)
     /**
      * @param PersistEvent $event
      */
-    public function handlePersist(PersistEvent $event)
+    public function handlePersistDocument(PersistEvent $event)
+    {
+        /** @var ResourceSegmentBehavior $document */
```

Comment:
If it is not a `ResourceSegmentBehavior` the execution will stop within the next if statement.